src/sidebar/components/Annotation/AnnotationEditor.js

```diff
@@ -43,63 +47,86 @@ function AnnotationEditor({
   );
 
   const store = useStoreProxy();
-  const draft = store.getDraft(annotation);
   const group = store.getGroup(annotation.group);
 
-  if (!draft) {
-    // If there's no draft, we can't be in editing mode
-    return null;
-  }
-
   const shouldShowLicense =
     !draft.isPrivate && group && group.type !== 'private';
 
   const tags = draft.tags;
   const text = draft.text;
   const isEmpty = !text && !tags.length;
 
-  const onEditTags = ({ tags }) => {
-    store.createDraft(draft.annotation, { ...draft, tags });
-  };
+  const onEditTags = useCallback(
+    ({ tags }) => {
+      store.createDraft(draft.annotation, { ...draft, tags });
+    },
+    [draft, store]
+  );
 
   /**
    * Verify `newTag` has content and is not a duplicate; add the tag
    *
    * @param {string} newTag
    * @return {boolean} - `true` if tag is added
    */
-  const onAddTag = newTag => {
-    if (!newTag || tags.indexOf(newTag) >= 0) {
-      // don't add empty or duplicate tags
-      return false;
-    }
-    const tagList = [...tags, newTag];
-    // Update the tag locally for the suggested-tag list
-    tagsService.store(tagList);
-    onEditTags({ tags: tagList });
-    return true;
-  };
+  const onAddTag = useCallback(
+    newTag => {
+      if (!newTag || tags.indexOf(newTag) >= 0) {
+        // don't add empty or duplicate tags
+        return false;
+      }
+      const tagList = [...tags, newTag];
+      // Update the tag locally for the suggested-tag list
+      tagsService.store(tagList);
+      onEditTags({ tags: tagList });
+      return true;
+    },
+    [onEditTags, tags, tagsService]
+  );
 
   /**
    * Remove a tag from the annotation.
    *
    * @param {string} tag
    * @return {boolean} - `true` if tag extant and removed
    */
-  const onRemoveTag = tag => {
-    const newTagList = [...tags]; // make a copy
-    const index = newTagList.indexOf(tag);
-    if (index >= 0) {
-      newTagList.splice(index, 1);
-      onEditTags({ tags: newTagList });
-      return true;
-    }
-    return false;
-  };
+  const onRemoveTag = useCallback(
+    tag => {
+      const newTagList = [...tags]; // make a copy
+      const index = newTagList.indexOf(tag);
+      if (index >= 0) {
+        newTagList.splice(index, 1);
+        onEditTags({ tags: newTagList });
+        return true;
+      }
+      return false;
+    },
+    [onEditTags, tags]
+  );
 
-  const onEditText = ({ text }) => {
-    store.createDraft(draft.annotation, { ...draft, text });
-  };
+  const onEditText = useCallback(
+    ({ text }) => {
+      store.createDraft(draft.annotation, { ...draft, text });
+    },
+    [draft, store]
+  );
+
+  /**
+   * @param {'private'|'shared'} isPrivate
```

Comment:
There is an asymmetry between the `isPrivate` prop (boolean) and the `onChangePrivacy` setter which takes an enum. I think I would be inclined to use a boolean in both cases, since we have no plans to add additional privacy levels within a group.

Also as a stylistic preference, I recommend putting the JSDoc on one line for a callback that only has one parameter and no other information.